services/production-tools/templates/deployment.yaml

```diff
@@ -0,0 +1,110 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "production-tools.fullname" . }}
+  labels:
+    {{- include "production-tools.labels" . | nindent 4 }}
+spec:
+  replicas: {{ .Values.replicaCount }}
+  selector:
+    matchLabels:
+      {{- include "production-tools.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      {{- with .Values.podAnnotations }}
+      annotations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      labels:
+        {{- include "production-tools.selectorLabels" . | nindent 8 }}
+    spec:
+      automountServiceAccountToken: false
+      imagePullSecrets:
+        - "pull-secret"
+      securityContext:
+        runAsNonRoot: true
+        runAsUser: 1000
+        runAsGroup: 1000
+      volumes:
+        # butler-secrets-raw is the secrets we get from vault
+        - name: "butler-secrets-raw"
+          secret:
+            secretName: "butler-secret"
+        # butler-secrets are the copied and chmoded versions
+        - name: "butler-secrets"
+          emptyDir: {}
+        - name: "cache-dir"
+          emptyDir: {}
+      # Have to fix permissions on the pgpass file.
+      # init container pattern borrowed from vo-cutouts.
+      initContainers:
+        - name: fix-secret-permissions
+          image: {{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}
+          imagePullPolicy: Always
+          command:
+            - "/bin/bash"
+            - "-c"
+            - |
+              cp -RL /home/worker/secrets-raw/* /home/worker/.lsst/
+              chown worker:worker /home/worker/.lsst/*
+              chmod 0400 /home/worker/.lsst/*
+          securityContext:
+            runAsNonRoot: false
+            runAsUser: 0
+            runAsGroup: 0
```

Comment:
I realized in a subsequent iteration of the vo-cutouts chart that you can delete the `chown` line (but keep the other two) and use the standard `securityContext` section of:
```
          securityContext:
            allowPrivilegeEscalation: false
            capabilities:
              drop:
                - "all"
```
The init container doesn't have to run as root because the files will be mounted world-readable. It can run as the regular user and the act of copying them will fix the ownership, which will make PostgreSQL happy (along with the `chmod`).